src/Confluent.Kafka/Serialization/StringSerializer.cs

```diff
@@ -33,25 +38,88 @@ public class StringSerializer : ISerializer<string>
         /// </param>
         public StringSerializer(Encoding encoding)
         {
-            this.encoding = encoding;
+            this.keyEncoding = encoding;
+            this.valueEncoding = encoding;
+        }
+
+        private static Encoding StringToEncoding(string s) 
+        {
+            switch (s)
+            {
+                case "UTF8":
+                    return Encoding.UTF8;
+                case "UTF7":
+                    return Encoding.UTF7;
+                case "UTF32":
+                    return Encoding.UTF32;
+                case "ASCII":
+                    return Encoding.ASCII;
+                case "Unicode":
+                    return Encoding.Unicode;
+                case "BigEndignUnicode":
+                    return Encoding.BigEndianUnicode;
+                default:
+                    throw new ArgumentException("unknown string encoding: " + s);
+            }
+        }
+
+        /// <summary>
+        ///     Initializes a new StringSerializer class instance.
+        /// </summary>
+        /// <param name="config">
+        ///     Configuration properties.
+        /// </param>
+        public StringSerializer(IEnumerable<KeyValuePair<string, object>> config)
+        {
+            if (config.Count(ci => ci.Key == "string.serializer.encoding.key") > 0)
```

Comment:
this is possibly unacceptable and making me think something more similar to java would be better - have a `Configure` method on `ISerializer`, but rather than return void, this could return a new config dictionary with the config property the serializer is interested in removed. there is a minor problem if two serializers want to pick up on the same property e.g. 'schema.registry.url'. If the calls to `Configure` are chained, the property will disappear, but the original dict could be used for both then the returned results merged.

From a user perspective, i think it's slightly worse (I prefer it being explicit that the config dict is being used to configure the serializers). It should probably be a runtime error if the user tries to configure both in the serializer constructor and via a config parameter.

Ok. I'll do another PR with the above variant and see how it looks.